stdlib/3/typing.pyi

```diff
@@ -178,7 +178,18 @@ class Container(Generic[_T_co]):
     @abstractmethod
     def __contains__(self, x: object) -> bool: ...
 
-class Sequence(Iterable[_T_co], Container[_T_co], Sized, Reversible[_T_co], Generic[_T_co]):
+if sys.version_info >= (3, 6):
+    class Collection(Sized, Iterable[_T_co], Container[_T_co], Generic[_T_co]):
+        pass
+
+if sys.version_info >= (3, 6):
+    class _Sequence(Collection[_T_co], Reversible[_T_co], Generic[_T_co]):
```

Comment:
@gvanrossum  It would have been even nicer to write this:

```
class Collection(Sized, Iterable[_T_co], Container[_T_co], Generic[_T_co]):
    pass

class Sequence(Collection[_T_co], Reversible[_T_co], Generic[_T_co]):
    ...

if sys.version_info >= (3, 6):
    del Collection
```

But for some reason I didn't understand, it didn't work (`Collection` remained accessible even when deleted from `typing.pyi`).

So I've updated the PR to use the `_Collection` approach.